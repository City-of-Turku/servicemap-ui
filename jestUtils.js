import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });