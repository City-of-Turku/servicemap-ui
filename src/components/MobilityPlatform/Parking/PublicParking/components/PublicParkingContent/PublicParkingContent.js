import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';
import { useIntl } from 'react-intl';
import { StyledContainer, StyledHeaderContainer, StyledTextContainer } from '../../../../styled/styled';
import TextComponent from '../../../../TextComponent';

const PublicParkingContent = ({ item }) => {
  const intl = useIntl();

  const renderText = (msgId, value) => (
    <StyledTextContainer>
      <Typography variant="body2">
        {value ? intl.formatMessage({ id: msgId }, { value }) : intl.formatMessage({ id: msgId })}
      </Typography>
    </StyledTextContainer>
  );

  const renderAccessInfo = accessValue => {
    const accessValueLower = accessValue.toLowerCase();
    if (accessValueLower === 'vapaa paasy') {
      return renderText('mobilityPlatform.content.publicParking.access');
    }
    if (accessValueLower === 'portti' || accessValueLower === 'portti, joka on auki') {
      return renderText('mobilityPlatform.content.publicParking.access.gate');
    }
    if (accessValueLower === 'puomi') {
      return renderText('mobilityPlatform.content.publicParking.access.barrier');
    }
    return null;
  };

  const translations = {
    placesTotal: 'mobilityPlatform.content.disabledParking.amount',
    totalTime: 'mobilityPlatform.content.publicParking.totalTime',
  };

  const names = {
    fi: item.name_fi,
    en: item.name_en,
    sv: item.name_sv,
  };

  const addressObj = {
    fi: item.address_fi,
    en: item.address_en,
    sv: item.address_sv,
  };

  return (
    <StyledContainer>
      <StyledHeaderContainer>
        <TextComponent textObj={names} isTitle />
      </StyledHeaderContainer>
      <div>
        <TextComponent messageId="mobilityPlatform.content.address" textObj={addressObj} />
        {renderText(translations.placesTotal, item.extra.paikkoja_y)}
        {item.extra.max_aika_h ? renderText(translations.totalTime, item.extra.max_aika_h) : null}
        {item.extra.rajoitustyyppi ? <TextComponent textObj={item.extra.rajoitustyyppi} /> : null}
        {item.extra.rajoit_lisat ? <TextComponent textObj={item.extra.rajoit_lisat} /> : null}
        {renderAccessInfo(item.extra.saavutettavuus.fi)}
      </div>
    </StyledContainer>
  );
};

PublicParkingContent.propTypes = {
  item: PropTypes.shape({
    address_fi: PropTypes.string,
    address_en: PropTypes.string,
    address_sv: PropTypes.string,
    name_fi: PropTypes.string,
    name_en: PropTypes.string,
    name_sv: PropTypes.string,
    extra: PropTypes.shape({
      paikkoja_y: PropTypes.number,
      max_aika_h: PropTypes.number,
      rajoitustyyppi: PropTypes.objectOf(PropTypes.string),
      rajoit_lisat: PropTypes.objectOf(PropTypes.string),
      saavutettavuus: PropTypes.objectOf(PropTypes.string),
    }),
  }),
};

PublicParkingContent.defaultProps = {
  item: {},
};

export default PublicParkingContent;
