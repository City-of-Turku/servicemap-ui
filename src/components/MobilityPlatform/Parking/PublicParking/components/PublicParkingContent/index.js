import PublicParkingContent from './PublicParkingContent';

export default PublicParkingContent;
