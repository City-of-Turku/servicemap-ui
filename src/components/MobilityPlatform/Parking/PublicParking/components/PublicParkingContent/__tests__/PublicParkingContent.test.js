/* eslint-disable max-len */
// Link.react.test.js
import React from 'react';
import { getRenderWithProviders } from '../../../../../../../../jestUtils';
import finnishTranslations from '../../../../../../../i18n/fi';
import { initialState } from '../../../../../../../redux/reducers/user';
import PublicParkingContent from '../index';

const mockProps = {
  item: {
    name_fi: 'Parkkipaikka',
    address_fi: 'Testikatu 1',
    extra: {
      paikkoja_y: 25,
      max_aika_h: 3.0,
      rajoitustyyppi: {
        fi: 'Erityisalue',
      },
      rajoit_lisat: {
        fi: 'Kiekkopysäköinti',
      },
      saavutettavuus: {
        fi: 'vapaa paasy',
      },
    },
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<PublicParkingContent />', () => {
  it('does match snapshot', () => {
    const { container } = renderWithProviders(<PublicParkingContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<PublicParkingContent {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(mockProps.item.name_fi);
    expect(p[1].textContent).toContain(`Osoite: ${mockProps.item.address_fi}`);
    expect(p[2].textContent).toContain(`Parkkipaikkojen määrä: ${mockProps.item.extra.paikkoja_y}`);
    expect(p[3].textContent).toContain(`Pysäköinnin enimmäisaika: ${mockProps.item.extra.max_aika_h}`);
    expect(p[4].textContent).toContain(mockProps.item.extra.rajoitustyyppi.fi);
    expect(p[5].textContent).toContain(mockProps.item.extra.rajoit_lisat.fi);
    expect(p[6].textContent).toContain(finnishTranslations['mobilityPlatform.content.publicParking.access']);
  });
});
