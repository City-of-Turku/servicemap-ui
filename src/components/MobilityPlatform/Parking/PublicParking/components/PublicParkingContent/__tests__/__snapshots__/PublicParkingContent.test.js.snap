// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublicParkingContent /> does match snapshot 1`] = `
<div>
  <div
    class="injectIntl(PublicParkingContent)-container-1"
  >
    <div
      class="injectIntl(PublicParkingContent)-headerContainer-2"
    >
      <div
        class="injectIntl(TextComponent)-margin-5"
      >
        <p
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          Parkkipaikka
        </p>
      </div>
    </div>
    <div
      class="injectIntl(PublicParkingContent)-textContainer-3"
    >
      <div
        class="injectIntl(TextComponent)-margin-5"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Osoite: Testikatu 1
        </p>
      </div>
      <div
        class="injectIntl(PublicParkingContent)-margin-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Parkkipaikkojen määrä: 25
        </p>
      </div>
      <div
        class="injectIntl(PublicParkingContent)-margin-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Pysäköinnin enimmäisaika: 3 tuntia
        </p>
      </div>
      <div
        class="injectIntl(TextComponent)-margin-5"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Erityisalue
        </p>
      </div>
      <div
        class="injectIntl(TextComponent)-margin-5"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Kiekkopysäköinti
        </p>
      </div>
      <div
        class="injectIntl(PublicParkingContent)-margin-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Esteetön pääsy
        </p>
      </div>
    </div>
  </div>
</div>
`;
